src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -104,7 +104,7 @@ public void finalized(AbstractBuild r) {
         AbstractProject<?, ?> project = r.getProject();
         Result result = r.getResult();
         AbstractBuild<?, ?> previousBuild = project.getLastBuild();
-        if (null != previousBuild) {
+        if (null != previousBuild && !r.isBuilding()) {
```

Comment:
I'm also troubled by this, I noticed it only now that I was working with #445 . My build job is a simple shell execution of:
```
sleep 2s
exit 0
```

And I keep getting 2 notifications when the build completes, which (I assume) are because `finalized` is fired very early, while jenkins hasn't marked up the build as finished. I looked into the jenkins source code, `Run.java`: 
https://github.com/jenkinsci/jenkins/blob/bcc4e4fe9c3dccc5ca74299d24f8f992c9d5cd71/core/src/main/java/hudson/model/Run.java#L737-L745
```
    public @Nonnull String getDurationString() {
        if (hasntStartedYet()) {
            return Messages.Run_NotStartedYet();
        } else if (isBuilding()) {
            return Messages.Run_InProgressDuration(
                    Util.getTimeSpanString(System.currentTimeMillis()-startTime));
        }
        return Util.getTimeSpanString(duration);
    }
```
And `Messages.properties`:
https://github.com/jenkinsci/jenkins/blob/584c34cda584b98b71d82d038e7d5ebe38534e18/core/src/main/resources/hudson/model/Messages.properties#L233
```
Run.InProgressDuration={0} and counting
```

And confirmed that the output I get, `test - #23 Success after 12 sec and counting (Open)` comes up because the `isBuilding()` is still `true`. 

My jenkins env is:
```
Virtual box machine
Debian GNU/Linux 9.6 (stretch)
jenkins: 2.155
```

I get the duplicate messages in our current master as well:
```
test - #41 Started by user admin (Open)
test - #41 Success after 4.1 sec and counting (Open)
test - #41 Success after 4.9 sec (Open)
```

Reply:
@KreAch3R I use JobDSL for creation job. -there is no duplicate.
Manually created : also only 2 messages in slack room;
```
test_misc » test_manuakl_slack - #1 Started by user Denis Shvedchenko (Open)
custom general mssage

test_misc » test_manuakl_slack - #1 Success after 0.3 sec and counting (Open)
custom success message
```